tests/system/src/test/java/test/robot/javafx/stage/StageAttributesTest.java

```diff
@@ -0,0 +1,325 @@
+import javafx.scene.layout.Pane;
+import javafx.scene.paint.Color;
+
+import test.util.Util;
+import test.robot.testharness.VisualTestBase;
+
+import static test.util.Util.TIMEOUT;
+
+public class StageAttributesTest extends VisualTestBase {
+
+    private static final int WIDTH = 400;
+    private static final int HEIGHT = 400;
+
+    private static final Color BOTTOM_COLOR = Color.LIME;
+    private static final Color TOP_COLOR = Color.RED;
+
+    private static final double TOLERANCE = 0.07;
+
+    private Stage bottomStage;
+    private Scene topScene;
+    private Stage topStage;
+
+    private void setupStages(boolean overlayed, boolean topShown) throws InterruptedException {
+        final CountDownLatch bottomShownLatch = new CountDownLatch(1);
+        final CountDownLatch topShownLatch = new CountDownLatch(1);
+
+        runAndWait(() -> {
+            // Bottom stage, should be visible after top stage is iconified
+            bottomStage = getStage(false);
+            bottomStage.initStyle(StageStyle.DECORATED);
+            Scene bottomScene = new Scene(new Pane(), WIDTH, HEIGHT);
+            bottomScene.setFill(BOTTOM_COLOR);
+            bottomStage.setScene(bottomScene);
+            bottomStage.setX(0);
+            bottomStage.setY(0);
+            bottomStage.setOnShown(e -> Platform.runLater(bottomShownLatch::countDown));
+            bottomStage.show();
+        });
+
+        assertTrue("Timeout waiting for bottom stage to be shown",
+            bottomShownLatch.await(TIMEOUT, TimeUnit.MILLISECONDS));
+
+        runAndWait(() -> {
+            // Top stage, will be inconified
+            topStage = getStage(true);
+            topStage.initStyle(StageStyle.DECORATED);
+            topScene = new Scene(new Pane(), WIDTH, HEIGHT);
+            topScene.setFill(TOP_COLOR);
+            topStage.setScene(topScene);
+            if (overlayed) {
+                topStage.setX(0);
+                topStage.setY(0);
+            } else {
+                topStage.setX(WIDTH);
+                topStage.setY(HEIGHT);
+            }
+            if (topShown) {
+                topStage.setOnShown(e -> Platform.runLater(topShownLatch::countDown));
+                topStage.show();
+            }
+        });
+
+        if (topShown) {
+            assertTrue("Timeout waiting for top stage to be shown",
+                topShownLatch.await(TIMEOUT, TimeUnit.MILLISECONDS));
+        }
+    }
+
+    @Test
+    public void testIconifiedStage() throws InterruptedException {
+        // Skip on Linux due to:
+        //  - JDK-8316423
+        assumeTrue(!PlatformUtil.isLinux());
+
+        setupStages(true, true);
+
+        runAndWait(() -> {
+            Color color = getColor(200, 200);
+            assertColorEquals(TOP_COLOR, color, TOLERANCE);
+
+            topStage.setIconified(true);
+        });
+
+        // wait a bit to let window system animate the change
+        Util.waitForIdle(topScene);
```

Comment:
In looking at this more closely, this change should be reverted back to using `sleep(1000)`. `Util.waitForIdle` is intended to wait for scene graph changes to be propagated, not for platform windowing operations to settle down, so using it could lead to intermittent failures.